### PR TITLE
Add relative paths for helm value files

### PR DESCRIPTION
**Pull request by Komorebi-E**
## Description

- Adds locals to make the paths relative for the Helm values template and the module's helm values files.

- Adds the PR template to this repo.

- Will run `terraform fmt` before merge.

## Related Issue

[Update SG rule descriptions](https://github.com/teamleadercrm/skyscrapers/issues/217)

## Motivation and Context

Prevents TF runs always updating the content because the path is different on each operators host, even though the content hasn't changed.

## How Has This Been Tested?

Tested with TL staging `terraform/stacks/elasticsearch` repo and this commit value instead of release value of `2.3.3`.

### Before change

```
  + module.logs.local_file.helm_values_file
      id:          <computed>
      content:     "## Prometheus MatchLabel selectors\nprometheus:\n  labels:\n    app: prometheus\n    prometheus: k8s-monitor\n    role: alert-rules\n\namazonService:\n  enabled: true\n\
nelasticsearch-exporter:\n  resources:\n    limits:\n      cpu: 10m\n      memory: 40Mi\n    requests:\n      cpu: 5m\n      memory: 20Mi\n  es:\n    uri: https://vpc-TL-staging-log
s-5pwpdhk4stpjmwwa4bpvikhoqm.eu-west-1.es.amazonaws.com\n\nprometheus-cloudwatch-exporter:\n  resources:\n    limits:\n      memory: 300Mi\n    requests:\n      cpu: 5m\n      memory: 150Mi
\n  aws:\n    role: \"arn:aws:iam::706305148424:role/kube2iam/cloudwatch_es_TL_staging\"\n  config: |-\n    region: eu-west-1\n    period_seconds: 60\n    set_timestamp: false\n
metrics:\n    - aws_namespace: AWS/ES\n      aws_metric_name: FreeStorageSpace\n      aws_dimensions: [ClientId, DomainName]\n      aws_dimension_select:\n        DomainName: [TL-st
aging-logs]\n      aws_statistics: [Minimum, Maximum, Average, Sum]\n"
      filename:    "/home/adey/skyscrapers/repos/CUSTOMERS/TL/skyscrapers/terraform/stacks/elasticsearch/helm-values.yaml"

Plan: 1 to add, 0 to change, 0 to destroy.
```

### After change

```
Terraform will perform the following actions:

  + module.logs.local_file.helm_values_file
      id:          <computed>
      content:     "## Prometheus MatchLabel selectors\nprometheus:\n  labels:\n    app: prometheus\n    prometheus: k8s-monitor\n    role: alert-rules\n\namazonService:\n  enabled: true\n\nelasticsearch-exporter:\n  resources:\n    limits:\n      cpu: 10m\n      memory: 40Mi\n    requests:\n      cpu: 5m\n      memory: 20Mi\n  es:\n    uri: https://vpc-TL-staging-logs-5pwpdhk4stpjmwwa4bpvikhoqm.eu-west-1.es.amazonaws.com\n\nprometheus-cloudwatch-exporter:\n  resources:\n    limits:\n      memory: 300Mi\n    requests:\n      cpu: 5m\n      memory: 150Mi\n  aws:\n    role: \"arn:aws:iam::706305148424:role/kube2iam/cloudwatch_es_TL_staging\"\n  config: |-\n    region: eu-west-1\n    period_seconds: 60\n    set_timestamp: false\n    metrics:\n    - aws_namespace: AWS/ES\n      aws_metric_name: FreeStorageSpace\n      aws_dimensions: [ClientId, DomainName]\n      aws_dimension_select:\n        DomainName: [TL-staging-logs]\n      aws_statistics: [Minimum, Maximum, Average, Sum]\n"
      filename:    "helm-values.yaml"

Plan: 1 to add, 0 to change, 0 to destroy.
```

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: helm-values.tf [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
+  // +1 for removing the "/"
+  helm_values_filename = "${substr("${path.module}/helm-values.yaml", length(path.cwd) + 1, -1)}"
 }
 
 data "aws_region" "current" {}
 
 data "template_file" "helm_values" {
   count    = "${length(var.prometheus_labels) != 0 ? 1 : 0}"
-  template = "${file("${path.module}/templates/helm-values.tpl.yaml")}"
+  template = "${file("${local.template_filename}")}"
```
Comment (SamClinckspoor) [inline helm-values.tf]: why is this change needed? as far as I know template location doesn't result in a change of the plan.

File: helm-values.tf [hunk 2]
```diff
@@ -30,5 +34,5 @@ data "template_file" "prometheus_kv_mapping" {
 resource "local_file" "helm_values_file" {
   count    = "${length(var.prometheus_labels) != 0 ? 1 : 0}"
   content  = "${data.template_file.helm_values.rendered}"
-  filename = "${path.cwd}/helm-values.yaml"
+  filename = "${local.helm_values_filename}"
```
Comment (SamClinckspoor) [inline helm-values.tf]: Doesn't this work as well? ` filename = "helm-values.yaml"`

File: helm-values.tf (hunk 1 above)
Comment (Komorebi-E) [reply helm-values.tf]: I agree that this isn't needed after retesting with it as it was before.

File: helm-values.tf (hunk 2 above)
Comment (Komorebi-E) [reply helm-values.tf]: Yes, tested and I agree it doesn't need to be as complex as I was making it for the local path.
From the TF Plan: `filename:    "helm-values.yaml"`

**Review by SamClinckspoor — APPROVED** [review]

File: helm-values.tf [hunk 3]
```diff
@@ -30,5 +30,5 @@ data "template_file" "prometheus_kv_mapping" {
 resource "local_file" "helm_values_file" {
   count    = "${length(var.prometheus_labels) != 0 ? 1 : 0}"
   content  = "${data.template_file.helm_values.rendered}"
-  filename = "${path.cwd}/helm-values.yaml"
+  filename = "helm-values.yaml"
```
Comment (iuriaranda) [inline helm-values.tf:34]: I would keep using the old notation `"${path.cwd}/helm-values.yaml"` as it makes it explicit where the file will end up. If you remove the `${path.cwd}` part, this will be dependent on the Terraform default behavior, which poses two small issues:
- if Terraform changes the default behavior this will stop working
- you have to go and search in the Terraform documentation what's the default behavior, which I haven't been able to find so far 😛

**Review by iuriaranda — CHANGES_REQUESTED** [review]
I'm requesting changes, but of course I'm open to discuss this as I might be wrong 😄

File: helm-values.tf (hunk 3 above)
Comment (SamClinckspoor) [reply helm-values.tf:34]: the cwd approach does not work. it saves the current pwd in the state, which introduces unnecessary changes if a different person runs it.

The previous solution in this PR `"${substr("${path.module}/helm-values.yaml", length(path.cwd) + 1, -1)}"` also does not work, as that saves the file in the module folder under `.terraform`

I think the plain notation as it is now is the best. it always ends up in the current directory where you run terraform.

Comment (Komorebi-E) [reply helm-values.tf:34]: I've added documentation and a comment in the code to clarify this.

Comment (iuriaranda) [reply helm-values.tf:34]: ok, I was under the impression this was purely an esthetical change.

**Review by iuriaranda — APPROVED** [review]
